tools/integration/lib/compareDefinitions.js

```diff
@@ -0,0 +1,218 @@
+    }
+  }
+
+  if (type1 === 'array') {
+    return compareArrays(val1, val2, path)
+  }
+
+  if (type1 === 'object') {
+    const objectComparison = compareDocuments(val1, val2, ignoredKeys, path)
+    const overallResult = objectComparison.overallResult
+
+    if (Object.keys(objectComparison.differences).length === 0) {
+      return { result: 'parity' }
+    } else {
+      return {
+        result: overallResult,
+        differences: objectComparison.differences
+      }
+    }
+  }
+
+  return handleLargeArrays(val1, val2, path, 'inconclusive')
+}
+
+function handleLargeArrays(val1, val2, path, result) {
+  const MAX_ELEMENTS = 10
+  let diff = {}
+
+  if (Array.isArray(val1) && Array.isArray(val2)) {
+    const set1 = new Set(val1.map(item => (typeof item === 'string' ? item.toLowerCase() : JSON.stringify(item))))
+    const set2 = new Set(val2.map(item => (typeof item === 'string' ? item.toLowerCase() : JSON.stringify(item))))
+
+    if (isSuperset(set1, set2)) {
+      const addedElements = [...set1].filter(x => !set2.has(x))
+      diff.addedElements = addedElements.slice(0, MAX_ELEMENTS)
+      result = 'improvement'
+    } else if (isSubset(set1, set2)) {
+      const missingElements = [...set2].filter(x => !set1.has(x))
+      diff.missingElements = missingElements.slice(0, MAX_ELEMENTS)
+      result = 'regression'
+    } else {
+      diff.staging = val1.slice(0, MAX_ELEMENTS)
+      diff.production = val2.slice(0, MAX_ELEMENTS)
+      result = 'inconclusive'
+    }
+
+    if (val1.length > MAX_ELEMENTS || val2.length > MAX_ELEMENTS) {
+      diff.truncated = true
+    }
+  } else {
+    diff.staging = Array.isArray(val1) ? val1.slice(0, MAX_ELEMENTS) : val1
+    diff.production = Array.isArray(val2) ? val2.slice(0, MAX_ELEMENTS) : val2
+    if ((Array.isArray(val1) && val1.length > MAX_ELEMENTS) || (Array.isArray(val2) && val2.length > MAX_ELEMENTS)) {
+      diff.truncated = true
+    }
+  }
+
+  return {
+    result: result,
+    differences: {
+      [result]: [
+        {
+          field: path,
+          diff: diff
+        }
+      ]
+    }
+  }
+}
+
+function compareArrays(arr1, arr2, path) {
+  if (JSON.stringify(arr1).toLowerCase() === JSON.stringify(arr2).toLowerCase()) {
+    return { result: 'parity' }
+  }
+
+  return handleLargeArrays(arr1, arr2, path, 'inconclusive')
+}
+
+function shouldIgnore(path, ignoredKeys) {
+  return ignoredKeys.some(prefix => path.startsWith(prefix))
+}
+
+function isEmpty(value) {
+  if (value === null || value === undefined) return true
+  if (Array.isArray(value)) return value.length === 0
+  if (typeof value === 'object') return Object.keys(value).length === 0
+  return false
+}
+
+function isSuperset(setA, setB) {
+  for (let elem of setB) {
+    if (!setA.has(elem)) {
+      return false
+    }
+  }
+  return true
+}
+
+function isSubset(setA, setB) {
+  for (let elem of setA) {
+    if (!setB.has(elem)) {
+      return false
+    }
+  }
+  return true
```

Comment:
nit: similar to return isSuperset(setB, setA)?

Reply:
This question may make sense to @RomanIakovlev as he has more context.  Can you elaborate on what you are suggesting?  Is that in place of `return true`?  Does `isSuperSet` already exist?  Would you expect this to be a simple change?  I'm assuming it is simple since you report it as a nit.

Reply:
Definitely can be a separate PR. The logic seems to be similar to
```
function isSubset(setA, setB) {
  return isSuperset(setB, setA)
}
```